lib/rspec/core/reporter.rb

```diff
@@ -2,6 +2,15 @@ module RSpec::Core
   # A reporter will send notifications to listeners, usually formatters for the
   # spec suite run.
   class Reporter
+    # @private
+    NOTIFICATIONS =
+      [
+        :close, :deprecation, :deprecation_summary, :dump_failures, :dump_pending,
+        :dump_profile, :dump_summary, :example_failed, :example_group_finished,
+        :example_group_started, :example_passed, :example_pending, :example_started,
+        :message, :seed, :start, :start_dump, :stop
+      ]
```

Comment:
Also, when we add new lifecycle events in the future, it'll be easy to forget to add them to this list.  Can you think of a way to add some kind of spec that enforces that all lifecycle events are in this list and prevents us from adding a new event w/o adding it here?